lib/internal/process/permission.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const {
+  ObjectFreeze,
+  ArrayPrototypePush,
+} = primordials;
+
+const permission = internalBinding('permission');
+const { validateString, validateArray } = require('internal/validators');
+const path = require('path');
```

Comment:
Can you destructure out the path functions that are used in this file to reduce the chances of them being tampered with.